### PR TITLE
Move a number of singer sources to incremental

### PR DESCRIPTION
**Pull request by sherifnada**
## What

Allow incremental syncs on the following singer integrations: 
* intercom
* exchangeratesapi 
* zendesk
* stripe
* twilio
* marketo

For most of these, this will be the only allowable sync mode on many of their streams. 

## How
1. Uses sync mode overrides introduced to Singer base in #1273 for SInger integrations which do not declare incrementality in accordance with the Singer spec. For the Singer integrations that "properly" declare incrementality, this is not needed (since singer_helpers automatically format the catalog when this is case). 
1. Adds a hack to standard tests which skips the second incremental sync test for a hardcoded set of container images to get around the inclusive cursor comparison issue. 

All integration images have been published.

### CODE REVIEW

File: docs/integrations/sources/twilio.md [hunk 1]
```diff
@@ -9,7 +9,7 @@ This Source Connector is based on a [Singer Tap](https://github.com/transferwise
 ### Output schema
 
 This Source is capable of syncing the following core Streams:
-
+__
```
Comment (sherifnada) [inline docs/integrations/sources/twilio.md:12]:
```suggestion
_
```

**Comment by sherifnada** [comment]
marketo is failing because of flakiness in the API

File: airbyte-integrations/connectors/source-exchangeratesapi-singer/setup.py [hunk 2]
```diff
@@ -33,8 +33,10 @@
     package_data={"": ["*.json"]},
     install_requires=[
         "tap-exchangeratesapi==0.1.1",
-        "pydantic==1.6.1",
```
Comment (sherifnada) [inline airbyte-integrations/connectors/source-exchangeratesapi-singer/setup.py:36]: was not needed

File: airbyte-integrations/bases/base-singer/base_singer/singer_helpers.py [hunk 3]
```diff
@@ -103,7 +103,7 @@ def override_sync_modes(airbyte_stream: AirbyteStream, overrides: SyncModeInfo):
     airbyte_stream.source_defined_cursor = overrides.source_defined_cursor or False
     if overrides.supported_sync_modes:
         airbyte_stream.supported_sync_modes = overrides.supported_sync_modes
-    if overrides.default_cursor_field:
+    if overrides.default_cursor_field or overrides.default_cursor_field == []:
```
Comment (jrhizor) [inline airbyte-integrations/bases/base-singer/base_singer/singer_helpers.py:106]: Maybe check that the cursor field is not `None`? That seems more "correct" for an `Optional` than excluding case by case falsey values.

File: airbyte-integrations/connectors/source-exchangeratesapi-singer/build.gradle [hunk 4]
```diff
@@ -1,16 +1,17 @@
 plugins {
-    id 'java'
     id 'airbyte-python'
     id 'airbyte-docker'
     id 'airbyte-integration-test-java'
```
Comment (jrhizor) [inline airbyte-integrations/connectors/source-exchangeratesapi-singer/build.gradle:4]:
```suggestion
```

File: docs/integrations/sources/twilio.md [hunk 5]
```diff
@@ -9,7 +9,7 @@ This Source Connector is based on a [Singer Tap](https://github.com/transferwise
 ### Output schema
 
 This Source is capable of syncing the following core Streams:
-
+_
```
Comment (jrhizor) [inline docs/integrations/sources/twilio.md:12]:
```suggestion

```

File: airbyte-integrations/connectors/source-marketo-singer/source_marketo_singer/source.py [hunk 6]
```diff
@@ -56,10 +58,58 @@ def check(self, logger: AirbyteLogger, config_container: ConfigContainer) -> Air
...
+            "activities_share_content",
+            "campaigns",
+            "lists",
+            "programs",
+        ]
+
+        return {s: SyncModeInfo([SyncMode.incremental], True, []) for s in incremental_streams}
 
     def read_cmd(self, logger: AirbyteLogger, config_path: str, catalog_path: str, state_path: str = None) -> str:
         # We don't pass in state to force the tap to run in full refresh since this tap does not respect the replication-method flag.
```
Comment (jrhizor) [inline airbyte-integrations/connectors/source-marketo-singer/source_marketo_singer/source.py:113]:
```suggestion
```

**Review by jrhizor — APPROVED** [review]
Looks good. Only minor changes requested.

**Review by cgardens — APPROVED** [review]

File: airbyte-integrations/bases/base-singer/base_singer/singer_helpers.py (hunk 3 above)
Comment (sherifnada) [reply airbyte-integrations/bases/base-singer/base_singer/singer_helpers.py:106]: that's better, will do

**Pull request merged**